Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleTableView.cpp

```diff
@@ -209,20 +342,30 @@ namespace ShaderManagementConsole
         const auto optionItem = horizontalHeaderItem(column);
         if (optionItem && !optionItem->text().isEmpty())
         {
-            if (const auto variantItem = item(row, column))
+            if (auto variantItem = item(row, column))
             {
+                QSignalBlocker blocker(this);
                 // Set or clear the option based on the item text
                 if (variantItem->text().isEmpty())
                 {
                     shaderVariant.m_options.erase(AZ::Name{optionItem->text().toUtf8().constData()});
+                    variantItem->setIcon(QIcon("D:/emptyoption.png"));
+                    variantItem->setToolTip(tr("runtime variable"));
                 }
                 else
                 {
                     shaderVariant.m_options[AZ::Name{optionItem->text().toUtf8().constData()}] = variantItem->text().toUtf8().constData();
+                    variantItem->setIcon({});
+                    variantItem->setToolTip("");
```

Comment:
Rather than using images to depict the empty value, one thing I had considered was just displaying the default value and changing the background color of the cell to denote whether or not it's overridden. This will allow the user to always see a value associated with the cell instead of having to guess what the default is.

Reply:
I'm not too sure; because the runtime is free to set any value, probably even encouraged to not let it be the default. If it's just in runtime mode to remain at default, it can be hardcoded in the shader and not be an option. Or maybe populate a variant row in the list with all explicit defaults for the same effect. The unset options are expected to have unpredictable values because their purpose is to have a C++ runtime free to set them on the fly according to game state.
I wonder if even, an option that doesn't explicitly receives a value from the runtime, and lets it be the default, shouldn't be a sort of "uninitialized error", at least a "validation layer" warning.